qsimcirq/qsim_circuit.py

```diff
@@ -15,6 +15,82 @@
 import numpy as np
 
 import cirq
+from qsimcirq import qsim
+
+
+def _cirq_gate_kind(gate):
+  if isinstance(gate, cirq.ops.identity.IdentityGate):
+    if gate.num_qubits == 1:
+      return qsim.kI
+    return qsim.kI2
+  if isinstance(gate, cirq.ops.XPowGate):
+    # cirq.rx also uses this path.
+    if gate.exponent == 1:
+      return qsim.kX
+    return qsim.kXPowGate
+  if isinstance(gate, cirq.ops.YPowGate):
+    # cirq.ry also uses this path.
+    if gate.exponent == 1:
+      return qsim.kY
+    return qsim.kYPowGate
+  if isinstance(gate, cirq.ops.ZPowGate):
+    # cirq.rz also uses this path.
+    if gate.exponent == 1:
+      return qsim.kZ
+    if gate.exponent == 0.5:
+      return qsim.kS
+    if gate.exponent == 0.25:
+      return qsim.kT
+    return qsim.kZPowGate
+  if isinstance(gate, cirq.ops.HPowGate):
+    if gate.exponent == 1:
+      return qsim.kH
+    return qsim.kHPowGate
+  if isinstance(gate, cirq.ops.CZPowGate):
+    if gate.exponent == 1:
+      return qsim.kCZ
+    return qsim.kCZPowGate
+  if isinstance(gate, cirq.ops.CXPowGate):
+    if gate.exponent == 1:
+      return qsim.kCX
+    return qsim.kCXPowGate
+  if isinstance(gate, cirq.ops.PhasedXPowGate):
+    return qsim.kPhasedXPowGate
+  if isinstance(gate, cirq.ops.PhasedXZGate):
+    return qsim.kPhasedXZGate
+  if isinstance(gate, cirq.ops.XXPowGate):
+    if gate.exponent == 1:
+      return qsim.kXX
+    return qsim.kXXPowGate
+  if isinstance(gate, cirq.ops.YYPowGate):
+    if gate.exponent == 1:
+      return qsim.kYY
+    return qsim.kYYPowGate
+  if isinstance(gate, cirq.ops.ZZPowGate):
+    if gate.exponent == 1:
+      return qsim.kZZ
+    return qsim.kZZPowGate
+  if isinstance(gate, cirq.ops.SwapPowGate):
+    if gate.exponent == 1:
+      return qsim.kSWAP
+    return qsim.kSwapPowGate
+  if isinstance(gate, cirq.ops.ISwapPowGate):
+    # cirq.riswap also uses this path.
+    if gate.exponent == 1:
+      return qsim.kISWAP
+    return qsim.kISwapPowGate
+  if isinstance(gate, cirq.ops.PhasedISwapPowGate):
+    # cirq.givens also uses this path.
+    return qsim.kPhasedISwapPowGate
+  if isinstance(gate, cirq.ops.FSimGate):
+    return qsim.kFSimGate
+  if isinstance(gate, cirq.ops.MatrixGate):
+    if gate.num_qubits == 1:
+      return qsim.kMatrixGate1
+    return qsim.kMatrixGate2
+  # No matches - decomposition is required.
+  # TODO: support decomposition.
+  return qsim.kGateDecomp
```

Comment:
`kGateDecomp` is a gate from Schmidt decomposition. It is internal to qsim. If there are no matches, an error should be raised.

Reply:
Done.